src/rand.rs

```diff
@@ -123,23 +120,82 @@ use self::sysrand_or_urandom::fill as fill_impl;
 #[cfg(any(target_os = "macos", target_os = "ios"))]
 use self::darwin::fill as fill_impl;
 
+#[cfg(target_os = "linux")]
+mod sysrand_chunk {
+    use error;
+    use c;
+
+    extern {
+        fn syscall(number: c::long, ...) -> c::long;
+    }
+
+    #[cfg(target_arch = "aarch64")]
+    const GETRANDOM: c::long = 278;
+    #[cfg(target_arch = "arm")]
+    const GETRANDOM: c::long = 384;
+    #[cfg(target_arch = "x86")]
+    const GETRANDOM: c::long = 355;
+    #[cfg(target_arch = "x86_64")]
+    const GETRANDOM: c::long = 318;
+
+    const EINTR: c::long = 4;
+
+    #[inline]
+    pub fn chunk(dest: &mut [u8]) -> Result<usize, error::Unspecified> {
+        let chunk_len: c::size_t = dest.len();
+        let flags: c::int = 0;
+        let r = unsafe {
+            syscall(GETRANDOM, dest.as_mut_ptr(), chunk_len, flags)
+        };
+        if r < 0 {
+            if r == EINTR {
+                // If an interrupt occurs while getrandom() is blocking
+                // to wait for the entropy pool, then EINTR is returned.
+                // Returning 0 will cause the caller to try again.
+                return Ok( 0 )
+            }
+            return Err(error::Unspecified);
+        }
+        Ok( r as usize )
+    }
+}
+
+#[cfg(windows)]
+mod sysrand_chunk {
+    use std::cmp::min;
+    use std::i32;
+    use error;
+
+    #[link(name = "Advapi32")]
+    extern "system" {
+        fn SystemFunction036( // also known as RtlGenRandom
```

Comment:
Please reformat this in the style that the rest of *ring* uses, something like:

```rust
extern "system" {
    fn SystemFunction036(random_buffer: *mut u8,
                         random_buffer_length: c::win32::ULONG)
                         -> c::win32::BOOLEAN;
}
```

Let's add a submodule `win32` to `ring::c` that defines the type aliases for ULONG and BOOLEAN. Note in particular that the infrastructure in `ring::c` allows us to test that our mapping of `ULONG` to `u32` and `BOOLEAN` to `u8` is correct on both 32-bit and 64-bit Windows.